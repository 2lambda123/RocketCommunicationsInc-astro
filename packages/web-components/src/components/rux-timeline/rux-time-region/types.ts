export enum PartialProperty {
    INSIDE = 'inside',
    LEFT = 'left',
    RIGHT = 'right',
}
