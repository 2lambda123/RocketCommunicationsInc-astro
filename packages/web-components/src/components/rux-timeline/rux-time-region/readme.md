# rux-time-region

<!-- Auto Generated Below -->


## Properties

| Property        | Attribute        | Description                                                                     | Type                                                                         | Default     |
| --------------- | ---------------- | ------------------------------------------------------------------------------- | ---------------------------------------------------------------------------- | ----------- |
| `end`           | `end`            | The end time                                                                    | `any`                                                                        | `undefined` |
| `hideTimestamp` | `hide-timestamp` | Optionally hide the bottom right timestamp.                                     | `boolean`                                                                    | `false`     |
| `selected`      | `selected`       | Visually displays the selected state                                            | `boolean`                                                                    | `false`     |
| `start`         | `start`          | The start time                                                                  | `any`                                                                        | `undefined` |
| `status`        | `status`         | Short hand attribute for displaying a Status icon and appropriate border color. | `"caution" \| "critical" \| "normal" \| "serious" \| "standby" \| undefined` | `undefined` |


## Slots

| Slot          | Description                    |
| ------------- | ------------------------------ |
| `"(default)"` | The content of the Time Region |


## Shadow Parts

| Part          | Description |
| ------------- | ----------- |
| `"container"` |             |


## Dependencies

### Depends on

- [rux-status](../../rux-status)

### Graph
```mermaid
graph TD;
  rux-time-region --> rux-status
  style rux-time-region fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
