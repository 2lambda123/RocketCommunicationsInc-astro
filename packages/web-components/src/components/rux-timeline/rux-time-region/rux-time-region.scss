:host {
    display: contents;
}

.rux-time-region {
    background: var(--color-background);
    border: 1px solid var(--color-status-normal-fill);
    // padding: 0.25rem;
    overflow: hidden;
    // display: flex;
    // flex-direction: column;
    min-height: 40px;
}

.rux-time-region__content {
    display: flex;
    align-items: center;
    font-family: var(--font-body-2-font-family);
    font-size: var(--font-body-2-font-size);
    font-weight: var(--font-body-2-font-weight);
    letter-spacing: var(--font-body-2-letter-spacing);
    ::slotted(rux-status) {
        margin-right: 0.25rem;
    }
}

.rux-time-region__datetime {
    margin-top: 0.25rem;
    text-align: right;
    font-family: var(--font-body-3-font-family);
    font-size: var(--font-body-3-font-size);
    font-weight: var(--font-body-3-font-weight);
    letter-spacing: var(--font-body-3-letter-spacing);
}

.hidden {
    display: none;
}
