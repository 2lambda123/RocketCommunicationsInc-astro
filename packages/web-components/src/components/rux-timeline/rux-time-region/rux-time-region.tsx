import { Element, Prop, Component, Host, h } from '@stencil/core'
import { formatInTimeZone } from 'date-fns-tz'

/**
 * @slot (default) - The content of the Time Region
 * @part container - The component's container
 * @part time-region-container - The container for time regions. Use this part to set a maximum height and enable vertical scrolling.
 * @part timestamp - The time region's timestamp
 */
@Component({
    tag: 'rux-time-region',
    styleUrl: 'rux-time-region.scss',
    shadow: true,
})
export class RuxTimeRegion {
    @Element() el!: HTMLRuxTimeRegionElement
    /**
     * The start date. Must be an ISO string "2021-02-02T05:00:00Z".
     */
    @Prop({ reflect: true }) start = ''
    /**
     * The end date. Must be an ISO string "2021-02-02T05:00:00Z"
     */
    @Prop({ reflect: true }) end = ''
    /**
     * Optionally hide the bottom right timestamp.
     */
    @Prop({ attribute: 'hide-timestamp' }) hideTimestamp = false
    /**
     * Short hand attribute for displaying a Status icon and appropriate border color.
     */
    @Prop() status?: 'normal' | 'critical' | 'serious' | 'caution' | 'standby'

    /**
     * Visually displays the selected state
     */
    @Prop() selected = false

    /**
     * @internal - The Time Regions's time zone. Set automatically from the parent Track component.
     */
    @Prop() timezone = 'UTC'

    get formattedTime() {
        if (!this.start || !this.end) {
            return false
        }

        try {
            return (
                formatInTimeZone(new Date(this.start), this.timezone, 'HH:mm') +
                '-' +
                formatInTimeZone(new Date(this.end), this.timezone, 'HH:mm')
            )
        } catch (e) {
            return false
        }
    }

    render() {
        return (
            <Host>
                <div
                    part="container"
                    class={{
                        'rux-time-region': true,
                        'rux-time-region--normal': this.status === 'normal',
                        'rux-time-region--critical': this.status === 'critical',
                        'rux-time-region--serious': this.status === 'serious',
                        'rux-time-region--caution': this.status === 'caution',
                        'rux-time-region--standby': this.status === 'standby',
                        'rux-time-region--selected': this.selected,
                    }}
                >
                    <div class="rux-time-region__content">
                        {this.status ? (
                            <rux-status
                                class={{
                                    'light-theme': this.selected,
                                }}
                                status={this.status}
                            ></rux-status>
                        ) : null}
                        <slot></slot>
                    </div>
                    {!this.hideTimestamp ? (
                        <div class="rux-time-region__datetime" part="timestamp">
                            {this.formattedTime}
                        </div>
                    ) : null}
                </div>
            </Host>
        )
    }
}
