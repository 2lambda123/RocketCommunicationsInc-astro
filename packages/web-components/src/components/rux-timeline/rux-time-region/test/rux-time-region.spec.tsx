import { newSpecPage } from '@stencil/core/testing'
import { RuxTimeRegion } from '../rux-time-region'

describe('rux-time-region', () => {
    it('renders', async () => {
        // const page = await newSpecPage({
        //     components: [RuxTimeRegion],
        //     html: `<rux-time-region></rux-time-region>`,
        // })

        // expect(page.root).toMatchSnapshot()

        expect(1).toBe(1)
    })
})
