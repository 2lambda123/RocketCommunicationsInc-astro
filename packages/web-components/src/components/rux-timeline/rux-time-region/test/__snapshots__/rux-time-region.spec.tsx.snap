// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rux-time-region renders 1`] = `
<rux-time-region end="" start="">
  <mock:shadow-root>
    <div class="rux-time-region" part="container">
      <div class="rux-time-region__content">
        <slot></slot>
      </div>
      <div class="rux-time-region__datetime" part="timestamp"></div>
    </div>
  </mock:shadow-root>
</rux-time-region>
`;
